consensus/packets/data_serializers.go

```diff
@@ -200,21 +200,31 @@ func (p1p *Phase1Packet) DeserializeWithoutHeader(data io.Reader, header *Packet
 	}
 
 	if p1p.hasSection2() {
-		claimsBuf, err := ioutil.ReadAll(data)
+		read, err := ioutil.ReadAll(data)
 		if err != nil {
 			return errors.Wrap(err, "[ Phase1Packet.DeserializeWithoutHeader ] Can't read Section 2")
 		}
-		claimsSize := len(claimsBuf) - 8
+		claimsSize := len(read) - SignatureLength
 
-		err = p1p.parseReferendumClaim(claimsBuf[:claimsSize])
+		err = p1p.parseReferendumClaim(read[:claimsSize])
 		if err != nil {
 			return errors.Wrap(err, "[ Phase1Packet.DeserializeWithoutHeader ] Can't parseReferendumClaim")
 		}
 
-		data = bytes.NewReader(claimsBuf[claimsSize:])
+		data = bytes.NewReader(read[claimsSize:])
+	} else {
+		read, err := ioutil.ReadAll(data)
+		if err != nil {
+			return errors.Wrap(err, "[ Phase1Packet.DeserializeWithoutHeader ] Can't read Section 2")
```

Comment:
we don't have section 2 here, fix error message

Reply:
done